internal/mptrie/store/update_and_query.go

```diff
@@ -3,57 +3,62 @@ package store
 import (
 	"encoding/base64"
 	"encoding/binary"
-	"encoding/json"
 	"errors"
 
+	"github.com/golang/protobuf/proto"
+
 	"github.com/syndtr/goleveldb/leveldb"
 
 	"github.com/IBM-Blockchain/bcdb-server/internal/mptrie"
 	"github.com/syndtr/goleveldb/leveldb/opt"
 )
 
 const (
-	Branch    = "B"
-	Extension = "E"
-	Value     = "V"
+	Branch    = 0x1
+	Extension = 0x2
+	Value     = 0x3
 )
 
 func (s *Store) GetNode(nodePtr []byte) (mptrie.TrieNode, error) {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 	key := base64.StdEncoding.EncodeToString(nodePtr)
-	nodeBytes, ok := s.inMemoryNodes[key]
+	storedNodeBytes, ok := s.inMemoryNodes[key]
 	if !ok {
-		nodeBytes, ok = s.nodesToPersist[key]
+		storedNodeBytes, ok = s.nodesToPersist[key]
 		if !ok {
-			storedNodeBytes, err := s.trieDataDB.Get(append(trieNodesNs, []byte(key)...), &opt.ReadOptions{})
+			var err error
+			storedNodeBytes, err = s.trieDataDB.Get(append(trieNodesNs, []byte(key)...), &opt.ReadOptions{})
 			if err != nil {
 				return nil, err
 			}
-			nodeBytes = &NodeBytesWithType{}
-			if err = json.Unmarshal(storedNodeBytes, nodeBytes); err != nil {
-				return nil, err
-			}
 		}
 	}
-	switch nodeBytes.NodeType {
+	nodeTypePrefix := storedNodeBytes[0]
+	switch nodeTypePrefix {
 	case Branch:
 		branchNode := &mptrie.BranchNode{
 			Children: make([][]byte, 16),
 		}
-		if err := json.Unmarshal(nodeBytes.NodeBytes, branchNode); err != nil {
+		if err := proto.Unmarshal(storedNodeBytes[1:], branchNode); err != nil {
 			return nil, err
 		}
+		// Strange bug - in array of byte slices (Children), proto.Unmarshal replace nil with empty byte slice
```

Comment:
just mention that proto3 converts nil to empty value rather than mentioning it as strange bug. https://github.com/golang/protobuf/issues/896